app/helpers/questions_helper.rb

```diff
@@ -0,0 +1,25 @@
+module QuestionsHelper
+  SORTING_OPTIONS = %w(All Week Month Year).freeze
+
+  def filtering(period)
+    return if period.nil?
+
+    if period == 'All'
+      @asked = Node.questions.to_a.size
+      @answered = Answer.all.map(&:node).uniq.size
+      "#{@asked} questions asked and #{@answered} questions answered"
+    else
+      @asked = Node.questions.where('created >= ?', 1.send(period.downcase).ago.to_i).to_a.size
+      @answered = Answer.where("created_at >= ?", 1.send(period.downcase).ago).map(&:node).uniq.size
+      "#{@asked} questions asked and #{@answered} questions answered in the past #{period}"
+    end
+  end
+
+  def options
+    SORTING_OPTIONS
+  end
+
+  def cache_quiz_stats
```

Comment:
This works, but then people coding do have to remember to use the "cached version". Another way would be to build this inside the `filtering()` method itself, so all calls to that are filtered. If people really want the non-filtered version, they can manually flush it with... i forget but something like `Rails.cache.flush(...)`. Make sense? This is exciting!

Reply:
Yes it does thanks, on it